reference/hoon-expressions/rune/buc.md

```diff
@@ -52,6 +51,84 @@ Irregular: `*` makes `%noun`, `^` makes `%cell`, `?` makes
 
 ## Runes
 
+
+### $| "bucbar"
+
+`[%bsbr p=spec q=hoon]`: structure that satisfies a validator.
+
+##### Expands to
+
+Still need help with this. From hoon.hoon:
+```hoon
+      ::  verify, $|
+      ::
+          {$bsbr *}
+        ^-  hoon
+        ::  push the raw product
+        ::
+        :+  %tsls  relative(mod p.mod)
+        ^-  hoon
+        ::  assert
+        ::
+        :+  %wtbn
+          ::  run the verifier
+          ::
+          [%cnhp [%tsbn $/3 q.mod] $/2]
+        ::  produce verified product
+        ::
+        $/2
+```
+
+
+##### Syntax
+
+Regular: **2-fixed**
+
+##### Discussion
+
+`$|` takes two arguments, a mold and a Hoon expression that is typically a gate
+that is used to validate the structure produced by the mold to ensure that the noun
+has a certain shape. It crashes if the input fails the validation test.
+
+For example, the elements of a `set` are treated as being
+unordered, but the values will necessarily possess an order by where they are in
+the memory. Thus if every `set` is stored using the same order scheme then faster algorithms involving `set`s may be
+written. This is not the same thing as casting - it is forcing a
```

Comment:
Maybe worth mentioning explicitly, if you just insert things in random locations in the set tree, they might be missed or mistreated by the algorithms that are in place.